### PR TITLE
fix: [ISH-293] correctly handle secrets for an engine from vault

### PR DESCRIPTION
**Pull request by phanyzewski**
- added comments for clarity
- renamed defaultEngine to defaultSecret
- added helpers for getting secrets from a path
- added helper for getting metadata url from a path

### CODE REVIEW

File: cmd/vault-staging-k8s-secret/config.go [hunk 1]
```diff
@@ -28,7 +30,7 @@ func newConfig(lgr *zap.Logger) *config {
 	if len(req) > 0 {
 		lgr.Fatal("bad configuration", zap.Strings("missing environment variables", req))
 	}
-	lgr.Debug("vault-path to the default engine", zap.String("VAULT_SECRET", defaultEngine))
+	lgr.Debug("vault-path to the default engine", zap.String("VAULT_SECRET", defaultSecret))
```
Comment (stevenaldinger) [inline cmd/vault-staging-k8s-secret/config.go:33]:
```suggestion
	lgr.Debug("vault-path to the default secret", zap.String("VAULT_SECRET", defaultSecret))
```

File: cmd/vault-staging-k8s-secret/config.go [hunk 2]
```diff
@@ -40,8 +42,34 @@ func newConfig(lgr *zap.Logger) *config {
...
+		defaultSecretPath:  defaultSecret,
+		overrideSecretPath: override,
+		k8sNamespace:       k8sNamespace,
 	}
 }
+
+// translatePath is a helper that converts a path to a vault secret
+// to a truncated path needed for the vault api engines list function
+//
+// ie staging/applications/data/foo/dotenv -> staging/applications/metadata/foo
```
Comment (stevenaldinger) [inline cmd/vault-staging-k8s-secret/config.go:55]: <3 this comment, thx for not making me think

File: cmd/vault-staging-k8s-secret/main.go [hunk 3]
```diff
@@ -42,45 +42,64 @@ func run(ctx context.Context, lgr *zap.Logger) error {
...
-	var verifiedEngines []string
-	for _, p := range engines {
-		lgr.Debug("verifying engine exists for path", zap.String("path", p))
-		if _, err := vault.ListEngines(ctx, p); err != nil {
-			verifiedEngines = append(verifiedEngines, p)
-		} else {
-			lgr.Error("cannot verify engine exists at path", zap.Error(err))
+	var vaultSecrets []string
+	var err error
+	lgr.Debug("getting secrets for engine", zap.String("engine", cfg.engine))
```
Comment (stevenaldinger) [inline cmd/vault-staging-k8s-secret/main.go:52]: is `engine` still the right terminology here?

**Comment by stevenaldinger** [comment]
I've got a little confusion over which pieces are legitimately talking about the engine/mount and which are talking about secret paths but over all this looks fantastic

**Review by stevenaldinger — APPROVED** [review]

File: cmd/vault-staging-k8s-secret/main.go (hunk 3 above)
Comment (phanyzewski) [reply cmd/vault-staging-k8s-secret/main.go:52]: yes at this point in the code we're trying to validate the engine itself exists

**Pull request merged**